lib/SILOptimizer/Analysis/ProtocolConformanceAnalysis.cpp

```diff
@@ -51,7 +55,8 @@ class NominalTypeWalker : public ASTWalker {
         for (auto *conformance : e->getLocalConformances()) {
           if (isa<NormalProtocolConformance>(conformance)) {
             auto *proto = conformance->getProtocol();
-            if (proto->getEffectiveAccess() <= AccessLevel::Internal) {
+            if (isWholeModule ||
```

Comment:
Given that this is only run in WMO, this if is the equivalent of doing `true || ...`. Should I remove the if entirely or keep it in case this API is ever used in non-wmo modules?